src/main/java/co/rsk/bitcoinj/script/NoRedeemScriptParser.java

```diff
@@ -0,0 +1,54 @@
+package co.rsk.bitcoinj.script;
+
+import co.rsk.bitcoinj.core.BtcECKey;
+import co.rsk.bitcoinj.core.ScriptException;
+import co.rsk.bitcoinj.core.Sha256Hash;
+import java.util.List;
+
+public class NoRedeemScriptParser implements RedeemScriptParser {
+
+    @Override
+    public boolean isStandardMultiSig() {
+        return false;
+    }
+
+    @Override
+    public boolean isFastBridgeMultiSig() {
+        return false;
+    }
+
+    @Override
+    public boolean isNotMultiSig() {
+        return true;
+    }
+
+    @Override
+    public ScriptType getScriptType() {
+        return ScriptType.UNDEFINED;
+    }
+
+    @Override
+    public int getM() {
+        return -1;
+    }
+
+    @Override
+    public int getSigInsertionIndex(Sha256Hash hash, BtcECKey signingKey) {
+        return 0;
+    }
+
+    @Override
+    public int findKeyInRedeem(BtcECKey key) {
+        return -1;
+    }
+
+    @Override
+    public List<BtcECKey> getPubKeys() {
+        throw new ScriptException("Only usable for multisig scripts.");
```

Comment:
Perhaps you could also throw this exception in the rest of the methods that do not apply, instead of returning values like `0` or `-1`

Reply:
that would be a breaking change